lighthouse-core/audits/seo/font-size.js

```diff
@@ -0,0 +1,286 @@
+  if (styleDeclaration && styleDeclaration.type === CSSStyleDeclaration.Type.Regular) {
+    const startLine = styleDeclaration.range ? styleDeclaration.range.startLine : 0;
+    const startColumn = styleDeclaration.range ? styleDeclaration.range.startColumn : 0;
+    return `${styleDeclaration.styleSheetId}@${startLine}:${startColumn}`;
+  } else {
+    return `node_${node.nodeId}`;
+  }
+}
+
+class FontSize extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'font-size',
+      description: 'Document uses legible font sizes.',
+      failureDescription: 'Document doesn\'t use legible font sizes.',
+      helpText: 'Font sizes less than 16px are too small to be legible and require mobile ' +
+      'visitors to “pinch to zoom” in order to read. ' +
+      '[Learn more](https://developers.google.com/speed/docs/insights/UseLegibleFontSizes).',
+      requiredArtifacts: ['FontSize', 'URL', 'Viewport'],
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const hasViewportSet = ViewportAudit.audit(artifacts).rawValue;
+    if (!hasViewportSet) {
+      return {
+        rawValue: false,
+        debugString: 'Text is illegible because of a missing viewport config',
+      };
+    }
+
+    const {
+      analyzedFailingNodesData,
+      analyzedFailingTextLength,
+      failingTextLength,
+      visitedTextLength,
+      totalTextLength,
+    } = artifacts.FontSize;
+
+    if (totalTextLength === 0) {
+      return {
+        rawValue: true,
+      };
+    }
+
+    const failingRules = getUniqueFailingRules(analyzedFailingNodesData);
+    const percentageOfPassingText =
+      (visitedTextLength - failingTextLength) / visitedTextLength * 100;
+    const pageUrl = artifacts.URL.finalUrl;
+
+    const headings = [
+      {key: 'source', itemType: 'url', text: 'Source'},
+      {key: 'selector', itemType: 'code', text: 'Selector'},
+      {key: 'coverage', itemType: 'text', text: '% of Page Text'},
+      {key: 'fontSize', itemType: 'text', text: 'Font Size'},
+    ];
+
+    const tableData = failingRules.sort((a, b) => b.textLength - a.textLength)
+      .map(({cssRule, textLength, fontSize, node}) => {
+        const percentageOfAffectedText = textLength / visitedTextLength * 100;
+        const origin = findStyleRuleSource(pageUrl, cssRule, node);
+
+        return {
+          source: origin.source,
+          selector: origin.selector,
+          coverage: `${percentageOfAffectedText.toFixed(2)}%`,
+          fontSize: `${fontSize}px`,
+        };
+      });
+
+    // all failing nodes which were not fully analyzed will be shown in the 'Other' category
+    if (analyzedFailingTextLength < failingTextLength) {
+      const percentageOfUnanalyzedFailingText =
+        (failingTextLength - analyzedFailingTextLength) / visitedTextLength * 100;
+
+      tableData.push({
+        source: 'Addtl illegible text',
```

Comment:
abbreviation bikeshed!!

https://writingexplained.org/english-abbreviations/additional kinda recommends `Add'l`. That was the one I was expecting. let's do that.